src/actions/blockchain.ts

```diff
@@ -237,22 +241,53 @@ export const checkUserStateAndSell = () => async (dispatch: Dispatch<any>, getSt
     }
     // Check allowance amount for SELLTOKEN
     // if allowance is ok, skip
-    const tokenAllowance = await promisedTokenAllowance
-    if (tokenAllowance) {
-      dispatch(openModal({
-        modalName: 'ApprovalModal',
-        modalProps: {
-          header: `Confirm ${sellName} Token movement`,
-          // tslint:disable-next-line
-          body: `Confirmation: DutchX needs your permission to move your ${sellName} Tokens for this Auction - please check ${activeProvider}`,
-        },
-      }))
+    const [needSellTokenAllowance, needOWLTokenAllowance] = await promisedTokensAndOWLAllowances
+    if (needSellTokenAllowance) {
+      const promisedChoice: Promise<string> = new Promise((accept) => {
+        dispatch(openModal({
+          modalName: 'ApprovalModal',
+          modalProps: {
+            header: `Confirm ${sellName} Token movement`,
+            // tslint:disable-next-line
+            body: `Confirmation: DutchX needs your permission to move your ${sellName} Tokens for this Auction - please check ${activeProvider}`,
+            onClick: accept,
+          },
+        }))
+      })
+      const choice = await promisedChoice
+
+      await dispatch(approveTokens(choice, 'SELLTOKEN'))
     // Go straight to sell order if deposit && allowance both good
-    } else {
-      dispatch(submitSellOrder())
+    }
+    if (needOWLTokenAllowance) {
+      const promisedChoice: Promise<string> = new Promise((accept) => {
+        dispatch(openModal({
+          modalName: 'ApprovalModal',
+          modalProps: {
+            header: `Approving OWL use for fees`,
+            body: `You have OWL available in your linked wallet address. Would you like to use OWL to pay for half of the fees on the DutchX? Any fee reduction due to MGN is valid and applied before the fee calculation.`,
+            buttons: {
+              button1: {
+                buttonTitle1: 'Approve',
+                buttonDesc1: 'Choose this option to approve the use of OWL to pay for half of the fees on the DutchX',
+              },
+              button2: {
+                buttonTitle2: 'Disallow',
+                buttonDesc2: 'Choose this option if you do not want to use OWL',
+              },
+            },
+            onClick: accept,
+          },
+        }))
+      })
+      const choice = await promisedChoice
+
+      await dispatch(approveTokens(choice, 'OWLTOKEN'))
```

Comment:
Wait, does this mean, you still trigger approval (modal et al) if choice is `Disallow`?

Reply:
No, it doesn't, ignore me